src/main/java/hudson/plugins/dry/DryPublisher.java

```diff
@@ -162,25 +78,35 @@ public String getPattern() {
         return pattern;
     }
 
+    /**
+     * Sets the Ant file-set pattern of files to work with.
+     */
+    @DataBoundSetter
+    public void setPattern(final String pattern) {
+        this.pattern = pattern;
+    }
+
     @Override
     public Action getProjectAction(final AbstractProject<?, ?> project) {
         return new DryProjectAction(project);
     }
 
     @Override
-    public BuildResult perform(final AbstractBuild<?, ?> build, final PluginLogger logger) throws InterruptedException, IOException {
+    public BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws
+            InterruptedException, IOException {
         logger.log("Collecting duplicate code analysis files...");
 
-        FilesParser dryCollector = new FilesParser(PLUGIN_NAME, StringUtils.defaultIfEmpty(getPattern(), DEFAULT_DRY_PATTERN),
-                    new DuplicationParserRegistry(getNormalThreshold(), getHighThreshold(), build.getWorkspace().getRemote(),
-                            getDefaultEncoding()),
-                    shouldDetectModules(), isMavenBuild(build));
+        FilesParser dryCollector = new FilesParser(PLUGIN_NAME, StringUtils.defaultIfEmpty(getPattern(),
+                DEFAULT_DRY_PATTERN),
+                new DuplicationParserRegistry(getNormalThreshold(), getHighThreshold(), workspace.getRemote(),
+                        getDefaultEncoding()),
+                shouldDetectModules(), isMavenBuild(build));
```

Comment:
BTW there are some gratuitous whitespace changes here; could have been collapsed to a one-line change AFAICT, easier to review.